sympy/matrices/tests/test_matrices.py

```diff
@@ -2341,6 +2341,37 @@ def test_col_insert():
         l.insert(i, 4)
         assert flatten(zeros(3).col_insert(i, c4).row(0).tolist()) == l
 
+def test_row_and_column_insert_for_indices():
```

Comment:
I think the code is well-tested viz. `test_row_inserts()` and `test_col_inserts()`. The error tests could probably go in `test_errors()`.